SPECS/td-agent/td-agent.patch

```diff
@@ -0,0 +1,263 @@
++                                 "#{plugin_gem_path}/08-aws-sdk-kms-1.30.0.gem",
++                                 "#{plugin_gem_path}/09-aws-sdk-sqs-1.24.0.gem",
++                                 "#{plugin_gem_path}/10-aws-sdk-s3-1.61.1.gem",
++                                 "#{plugin_gem_path}/11-fluent-plugin-s3-1.3.0.gem",
++                                 "#{plugin_gem_path}/12-webhdfs-0.9.0.gem",
++                                 "#{plugin_gem_path}/13-fluent-plugin-webhdfs-1.2.4.gem",
++                                 "#{plugin_gem_path}/14-fluent-plugin-rewrite-tag-filter-2.3.0.gem",
++                                 "#{plugin_gem_path}/15-ruby-kafka-1.0.0.gem",
++                                 "#{plugin_gem_path}/rdkafka-0.7.0.gem",
++                                 "#{plugin_gem_path}/17-fluent-plugin-kafka-0.13.0.gem",
++                                 "#{plugin_gem_path}/18-elasticsearch-7.6.0.gem",
++                                 "#{plugin_gem_path}/19-fluent-plugin-elasticsearch-4.0.7.gem",
++                                 "#{plugin_gem_path}/20-prometheus-client-0.9.0.gem",
++                                 "#{plugin_gem_path}/21-fluent-plugin-prometheus-1.7.3.gem",
++                                 "#{plugin_gem_path}/22-fluent-plugin-prometheus_pushgateway-0.0.2.gem",
++                                 "#{plugin_gem_path}/23-fluent-plugin-record-modifier-2.1.0.gem",
++                                 "#{plugin_gem_path}/24-systemd-journal-1.3.3.gem",
++                                 "#{plugin_gem_path}/25-fluent-plugin-systemd-1.0.2.gem",
++                                 "#{plugin_gem_path}/26-nokogiri-1.11.0.rc2.gem"]
++            
++            install_gems(plugin_gems_files)
++          end
++        end
+       end
+ 
+       desc "Install fluentd"
+@@ -278,7 +335,9 @@ class BuildTask
+           archive_path = @download_task.file_fluentd_archive
+           fluentd_dir = archive_path.sub(/\.tar\.gz$/, '')
+           sh("tar", "xvf", archive_path) unless File.exists?(fluentd_dir)
++          cp("/usr/src/mariner/SOURCES/gemfile.tar.gz", "fluentd-#{FLUENTD_REVISION}")
+           cd("fluentd-#{FLUENTD_REVISION}") do
++            sh("tar", "xvf", "gemfile.tar.gz", "-C", ".")
+             sh("rake", "build")
+             gem_install("pkg/fluentd-*.gem")
+           end
+diff --git a/td-agent/core_gems.rb b/td-agent/core_gems.rb
+index b2592e4..3a3e026 100644
+--- a/td-agent/core_gems.rb
++++ b/td-agent/core_gems.rb
+@@ -1,24 +1,14 @@
+ dir 'core_gems'
+ 
+-download "bundler", "2.1.4"
+-download "msgpack", "1.3.3"
+-download "cool.io", "1.6.0"
+-download 'serverengine', '2.2.1'
+-download "oj", "3.10.6"
+-download "async-http", "0.50.13"
+-download "http_parser.rb", "0.6.0"
+-download "yajl-ruby", "1.4.1"
+-download "sigdump", "0.2.4"
+-download "tzinfo", "2.0.2"
+-download "tzinfo-data", "1.2019.3"
++"/usr/src/mariner/SOURCES/00-bundler-2.1.4.gem"
```

Comment:
Would it be possible to use our sources server for the binaries instead? Some of them go into multiple MBs and we should avoid keeping larger files in a git repo. @schmittjoseph: do you think it's possible (without any huge changes) to make the tooling try and download all sources missing locally from the server? Or maybe that's already happening?

Reply:
That's already happening. Any and all sources not found locally for a given spec will be retrieved from the `SOURCE_URL`.